source/base/function_parser.cc

```diff
@@ -53,6 +54,33 @@ FunctionParser<dim>::FunctionParser(const unsigned int n_components,
 {}
 
 
+template <int dim>
+FunctionParser<dim>::FunctionParser(const std::string &expression,
+                                    const std::string &constants,
+                                    const std::string &variable_names,
+                                    const double       h)
+  : AutoDerivativeFunction<dim>(
+      h,
+      Utilities::split_string_list(expression, ';').size())
+  , initialized(false)
+  , n_vars(0)
+{
+  auto constants_map = Patterns::Tools::Convert<ConstMap>::to_value(
+    constants,
+    Patterns::Map(Patterns::Anything(),
+                  Patterns::Double(),
+                  0,
+                  Patterns::Map::max_int_value,
+                  ",",
+                  "=")
+      .clone());
```

Comment:
why do you need clone here?

Reply:
because the function expects a std::unique_ptr.

Reply:
I think what @tjhei was thinking of was to use `std_cxx14::make_unique<Patterns::Map>(...arguments as above...)`.